oio/cli/admin/service_check.py

```diff
@@ -0,0 +1,231 @@
+# Copyright (C) 2019 OpenIO SAS, as part of OpenIO SDS
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from cliff import lister
+
+from oio.conscience.client import ConscienceClient
+
+
+class BaseCheckCommand(lister.Lister):
+    """
+    Base class for all check commands.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super(BaseCheckCommand, self).__init__(*args, **kwargs)
+        self._zkcnxstr = None
+        self.catalog = None
+        self.live = None
+
+    def get_parser(self, prog_name):
+        parser = super(BaseCheckCommand, self).get_parser(prog_name)
+        parser.add_argument(
+            '--catalog',
+            type=str,
+            help="Load catalog from file."
+        )
+        return parser
+
+    def load_catalog(self, parsed_args):
+        # Load the live services
+        self.live = self.load_live_services()
+        self.live = tuple(self.live)
+        self.logger.info("Catalog: Loaded %d services", len(self.live))
+        for t, i, p, s in self.live:
+            self.logger.debug("live> %s %s %d score=%d", t, i, p, s)
+
+        # Load a catalog of expected services
+        self.catalog = list()
+        if parsed_args.catalog:
+            self.catalog = self.load_catalog_from_file(parsed_args.catalog)
+        else:
+            for t, i, p, s in self.live:
+                self.catalog.append((t, i, p, s))
+        self.catalog = tuple(self.catalog)
+        self.logger.info("Catalog: Loaded %d services", len(self.catalog))
+        for t, i, p, s in self.catalog:
+            self.logger.debug("catalog> %s %s %d", t, i, p)
+
+    @staticmethod
+    def filter_services(srv, srvtype):
+        for t, i, p, s in srv:
+            if t == srvtype:
+                yield t, i, p, s
+
+    def load_live_services(self):
+        client = ConscienceClient({"namespace": self.app.options.ns})
```

Comment:
I suggest using the `ConscienceClient` instance create by `ClientManager`, to avoid loading the local namespace configuration file when the CLI user specifies a proxy endpoint with `--oio-proxy`.
Something like:
```self.app.client_manager.conscience```